flytekit/core/type_engine.py

```diff
@@ -270,7 +270,7 @@ def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], exp
         return Literal(scalar=Scalar(generic=struct))
 
     def to_python_value(self, ctx: FlyteContext, lv: Literal, expected_python_type: Type[T]) -> T:
-        if not (lv and lv.scalar and lv.scalar.generic):
+        if not (lv and lv.scalar and lv.scalar.generic is not None):
             raise AssertionError("Can only covert a generic literal to a Protobuf")
```

Comment:
Can you also fix the typo?
```suggestion
            raise AssertionError("Can only convert a generic literal to a Protobuf")
```

Reply:
👍🏽